code/app.py

```diff
@@ -26,29 +30,39 @@
 def static_file(path):
     return app.send_static_file(path)
 
-
-@app.route("/api/config", methods=["GET"])
+AZURE_AUTH_TYPE = os.environ.get("AZURE_AUTH_TYPE", "keys")
+# Initialize Azure keys based on authentication type and environment settings. 
+# When AZURE_AUTH_TYPE is not 'rbac' and USE_KEY_VAULT environment variable is set, 
+# Azure keys are securely fetched from Azure Key Vault using DefaultAzureCredential. 
+# Otherwise, keys are obtained from environment variables, with fallbacks to None or 
+# an empty string depending on the AZURE_AUTH_TYPE.
+if not AZURE_AUTH_TYPE == 'rbac' and os.environ.get("USE_KEY_VAULT"):
+    credential = DefaultAzureCredential()
+    secret_client = SecretClient(os.environ.get("AZURE_KEY_VAULT_ENDPOINT"), credential)
+    AZURE_SEARCH_KEY = secret_client.get_secret(os.environ.get("AZURE_SEARCH_KEY")).value
```

Comment:
You might want to first make sure the environment variable exists and then try to fetch it, so that you don't try to fetch a secret called None. It's an error either way but catching earlier is easier debugging.

Reply:
Could we push this down to EnvHelper as this code is repeated there too

Reply:
In fact, in the [bicep file](https://github.com/jongio/chat-with-your-data-solution-accelerator/blob/main/infra/app/adminweb.bicep#L38-L44), all `AZURE_XXX_KEY` are associated with `USE_KEY_VAULT`, so `get_secret(None)` will not appear here.